src/util.h

```diff
@@ -79,6 +79,15 @@ bool RenameOver(fs::path src, fs::path dest);
 bool LockDirectory(const fs::path& directory, const std::string lockfile_name, bool probe_only=false);
 bool DirIsWritable(const fs::path& directory);
 
+//! Return the original FILE* unchanged. On POSIX systems that support it,
+//! also advise the kernel that the file will be accessed sequentially.
+FILE* AdviseSequential(FILE *file);
+
+//! Close a file and return the result of fclose(). On POSIX systems that
+//! support it, advise the kernel to remove the file contents from the page
+//! cache (which can help on memory-constrained systems).
+int CloseAndDiscard(FILE *file);
```

Comment:
"discard" sounds to me as if writes to the file will be discarded, too
maybe `CloseAndUncache`?